nucypher-core-python/src/lib.rs

```diff
@@ -630,6 +631,126 @@ impl ReencryptionResponse {
     }
 }
 
+//
+// Threshold Decryption Request
+//
+
+#[pyclass(module = "nucypher_core")]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct ThresholdDecryptionRequest {
+    backend: nucypher_core::ThresholdDecryptionRequest,
+}
+
+#[pymethods]
+impl ThresholdDecryptionRequest {
+    #[new]
+    pub fn new(
+        id: u16,
+        variant: u8,
+        ciphertext: &[u8], // TODO use ferveo Ciphertext type
+        conditions: Option<&Conditions>,
+        context: Option<&Context>,
+    ) -> Self {
+        let ferveo_variant: FerveoVariant;
+        match variant {
+            0 => ferveo_variant = FerveoVariant::SIMPLE,
+            1 => ferveo_variant = FerveoVariant::PRECOMPUTED,
+            _ => panic!("Invalid variant"),
```

Comment:
You can instead raise an exception in Python; see methods returning `PyResult`

Reply:
✔️